### DISCUSSION TITLE
is turning off deepequal check still supported?

### DISCUSSION THREAD

**Discussion by stefanprobst** [TanStack/query#893] [General] [answered]
[according to docs](https://react-query.tanstack.com/docs/guides/important-defaults), setting `config.isDataEqual = () => false` will disable the default deepequal check for query results.

however, it looks like in the current version, query results will always be deep compared *before* `isDataEqual` is run:
https://github.com/tannerlinsley/react-query/blob/master/src/core/query.ts#L292-L298

is there a way to avoid this?

**Comment by boschni** [comment, marked as answer]
Hi! The code you are referring to is to structurally share data between query results, which will indeed traverse through the results. We could create a flag to toggle this behaviour, but may I ask why you want to avoid it?

**Reply by stefanprobst** [reply]
i'm working with an unnecessarily large payload from an api i have no control over. so far, i have preferred to just re-render, rather than traversing that.

switching the deepequal default off is explicitly mentioned early in the docs, so it would be great if this was still possible (at least until v3)

thanks for your work on react-query!

**Reply by boschni** [reply]
There is now a `structuralSharing` query option which you can use to turn it off :)

**Reply by stefanprobst** [reply]
awesome, thanks!
